2025 Data Entry

jkleiber opened the pull request.

jkleiber commented:
@PChild @aidnem please check my work on the data entry form - it cannot submit yet, but I want the UI to be (mostly) correct before committing to making the database schema

While it cannot submit, you should try submit and reset to make sure this all works. If you submit it, it should show a JSON string with all your data entries

aidnem commented:
@jkleiber there's an interesting visual bug on MacOS firefox where the radio selector borders are brightly colored and not rounded:
<img width="677" alt="image" src="https://github.com/user-attachments/assets/f4f979f0-e505-45de-8a2d-8312a4b2d256" />

jkleiber commented:
@aidnem that's intended behavior. I have the form set to error if something isn't filled out and that's what it looks like for the radio buttons

aidnem commented:
Okay okay I see. I'm going to test it again but it looked like what it printed out in the console was the correct data and everything.
I'll read over the code but I assume it'll all be fine as well.

aidnem commented on src/lib/2025/match-scouting-form.ts:
```diff
@@ -88,19 +94,55 @@ export function getMatchScoutSchema() {
+                    options: {
+                        labels: ["Processor", "Net", "Dislodged"],
+                        sections: [
+                            { key: "success", text: "Succcess" },
```
Too many c's in success

jkleiber replied:
learned how to spell in most recent commit

aidnem commented on src/lib/2025/match-scouting-form.ts:
```diff
@@ -109,31 +151,161 @@ export function getMatchScoutSchema() {
+                    options: {
+                        labels: ["Processor", "Net", "Dislodged"],
+                        sections: [
+                            { key: "success", text: "Succcess" },
```
Too many c's in success

jkleiber replied:
learned how to spell in most recent commit

aidnem commented on src/lib/2025/match-scouting-form.ts:
```diff
@@ -88,19 +94,55 @@ export function getMatchScoutSchema() {
-                    required: false
+                    required: false,
+                    error: false
                 },
```
The only thing described in #67 that's missing from the UI is "Coopertition true/false" from this section.

aidnem reviewed (changes requested):
I've never written vue code before so I just glanced over the code since it worked.

The only issues I found are the 2 spelling nitpicks and the missing "Coopertition true/false" field in autonomous.

jkleiber commented:
@aidnem I'll definitely fix the spelling stuff

For the co-op field - what exactly is our motivation in having that? If 2 algae are scored in the processor at any point in the match then the team has done co-op on their own. Would we not simply detect that during data analysis?

aidnem commented:
@jkleiber Yeah you're right. Sorry I forgot there wasn't a button this year.

PChild commented:
Does it make more sense to have the scout name as a filterable dropdown?

Could we have the match number field auto-increment instead of resetting on form submission?

Could we do conditional hiding or value setting of form sections? If a robot no-shows we don't need to ask about any other performance stuff, if they don't move in auto we don't need to show any scoring fields (less sure about this one).

Is it worth adding a climb speed metric? I'm not sure how often we'd use it.

Can we increase space between scored and missed please?
![image](https://github.com/user-attachments/assets/8fd5c69f-ebd9-448f-943c-3c34390bf459)

PChild commented:
It also needs to handle wrapping on mobile:

![Screenshot_20250112-114226](https://github.com/user-attachments/assets/f0b21aec-b16a-464f-9e6f-2f127f366aed)

PChild commented:
> @aidnem I'll definitely fix the spelling stuff
> 
> For the co-op field - what exactly is our motivation in having that? If 2 algae are scored in the processor at any point in the match then the team has done co-op on their own. Would we not simply detect that during data analysis?

Also how much do we care about co-op? It only matters in quals.

jkleiber commented:
@PChild 

> Does it make more sense to have the scout name as a filterable dropdown?

I can do that. I don't have a list of names though so I'll create a ticket to do that.

> Could we have the match number field auto-increment instead of resetting on form submission?

Sure, I can add that feature in this PR.

> Could we do conditional hiding or value setting of form sections? If a robot no-shows we don't need to ask about any other performance stuff, if they don't move in auto we don't need to show any scoring fields (less sure about this one).

The relative effort to add this feature is high and the value (imo) is low in terms of priority (I would rather have the pick list page than this feature for example). The scouts can just scroll as needed for now. That said, it's a good feature request so I will make a ticket for this

> Is it worth adding a climb speed metric? I'm not sure how often we'd use it.

We can do another 1-5 scale for climb speed and remove it if it is dumb. I will add it to this PR.

> Can we increase space between scored and missed please?

Yes, I will do that in this PR.

>  it might be nice to have visual representations of coral and algae underneath the text

Agreed, but in a future PR. I will make a ticket for this

> It also needs to handle wrapping on mobile

yeah... I'll try to make this better

PChild commented:
Do we want reef and both options for algae intake in pit scouting? 

Is it worth asking about climb capabilities in pit scouting? We should default to match data but if a team says they can do something but haven't shown it it might be worth checking up on their climb if considering them for a second pick or something.

jkleiber commented:
@PChild 

In the most recent commits I:
- added climb type to the pit scouting form, and added algae intake reef and both as well
- made auto-increment for match number work (note: if you reset the form this value will not reset now)
- spaced out the counters more
- added `overflow: scroll` for the views so mobile works somewhat better

As for the co-op discussion, I don't think we care. It may be tactically interesting during quals matches but won't be relevant for playoffs

jkleiber commented:
@aidnem since you requested changes, I need you to approve this in order to merge
